bpfd/src/bpf.rs

```diff
@@ -266,17 +260,11 @@ impl BpfManager {
                 .await
                 .map_err(|e| BpfdError::Error(format!("can't create map dir: {e}")))?;
 
-            let program_bytes = if program
+            let program_bytes = program
                 .data
-                .path
-                .clone()
-                .contains(BYTECODE_IMAGE_CONTENT_STORE)
-            {
-                get_bytecode_from_image_store(program.data.path.clone()).await?
-            } else {
-                read(program.data.path.clone()).await?
-            };
-
+                .location
+                .get_program_bytes(&program.data.section_name)
```

Comment:
So here we **may** need to change the programData's section name based on the image right? 

I think `program_bytes` will actually need to be a method on the `ProgramData` stuct, which will then internally call the `Location.get_program_bytes()` and override `section_name` if needed.

Does that make sense?